### DISCUSSION TITLE
File IO from web example

### DISCUSSION THREAD

**Discussion by c-git** [emilk/egui#3783] [Ideas]
Hi,

I struggled for several hours to figure out how to do File IO in the WASM version of my application (in my browser). Eventually after much suffering I found https://github.com/woelper/egui_pick_file which worked. I want to modify it a bit to also support saving the file and disabling save/load while IO is in progress. Would this be a useful PR to add it to the examples. Wanted to find out before I proceed to do the PR as it will take me some amount of time to do it and test it.

**Comment by emilk** [comment]
It is a bit cumbersome to have freestanding examples that compile and run as wasm (or at least it is currently). Therefore I suggest you instead improve https://github.com/woelper/egui_pick_file and we can link to that as an example from the main `egui` repo

**Reply by c-git** [reply]
Ok that makes sense. And will also try to setup a demo site so users can try it and see if it meets their needs.

**Comment by c-git** [comment]
@woelper has updated the example to also show how to save. So it's ready to be linked if that is the way we want to go.

I've thought about it more and I'm not sure putting a link in the readme is actually the right way to go. I'm wondering if it would be better to add it to the web_demo or maybe the current example for dialogs could be kept and used to show how to do dragging of files onto the window but this one could show an alternate way that would work for both wasm and native even if the example is native only?

**Comment by c-git** [comment]
Created the PR https://github.com/emilk/egui/pull/3835
